app/src/main/java/ch/admin/bag/dp3t/reports/ReportsFragment.java

```diff
@@ -144,15 +144,14 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
 			if (tracingStatusInterface.isReportedAsInfected()) {
 				headerType = ReportsHeaderFragment.Type.POSITIVE_TESTED;
 				infectedView.setVisibility(View.VISIBLE);
+				
+				long oldestSharedKeyDateMillis = secureStorage.getPositiveReportOldestSharedKey();
+				if (oldestSharedKeyDateMillis > 0L) {
+					long displayDateMillis =
+							Math.max(oldestSharedKeyDateMillis, System.currentTimeMillis() - MAX_EXPOSURE_AGE_MILLIS);
```

Comment:
This logic is wrong. With this while being in isolation the date since when the keys were shared would potentially change over time. We need to do the max logic before we save the Date and only then.